### DISCUSSION TITLE
The issues encountered when using CWatM-MODFLOW

### DISCUSSION THREAD

**Discussion by weixiaoxun** [iiasa/CWatM#157] [Q&A] [answered]
![微信图片_20240919175202](https://github.com/user-attachments/assets/3c91e5e1-7f8d-4fec-a895-df13508a6f69)
Dear Professors @mikhailsmilovic @dof1985,

I hope this email finds you well.

I am currently working on an experiment using CWatM-MODFLOW for my study area, but I am encountering an error right at the beginning of the simulation (attached is an image of the error). I have tried adjusting the initial_water_table_depth parameter multiple times, but the issue persists.

Could you kindly provide some guidance on how to resolve this issue? I would greatly appreciate your assistance.

Thank you for your time and support.

Best regards,
weixiaoxun
henu

**Comment by mikhailsmilovic** [comment]
Guten Morgen, the message could relate to a few things. To start:

what version of xmipy is being used?

In the OPTIONS section, include: use_complex_solver_for_modflow = True

**Reply by weixiaoxun** [reply]
Guten Morgen @mikhailsmilovic 

Thank you very much for your prompt response to my query.

After checking the model's settings, I wasn't able to locate the keyword use_complex_solver_for_modflow in the options(figure 1). Additionally, I wanted to confirm that the version of XMIPY I am using is 1.0.0(figure 2).

Could you kindly advise on the next steps I should take to resolve this issue? Your guidance has been invaluable, and I greatly appreciate your time and expertise.

Best regards,
wei xiaoxun
HENU
![微信图片_20240920164044](https://github.com/user-attachments/assets/9e26cd22-aac5-4f6a-b117-5f676e0b6833)
![微信截图_20240920165337](https://github.com/user-attachments/assets/27ecaeec-a8d7-415e-a175-aece9e4bce5c)

**Comment by dof1985** [comment, marked as answer]
Dear @weixiaoxun 

As mentioned by @mikhailsmilovic you'll need to add the following line in your options section in the settings file:
use_complex_solver_for_modflow=True

If it still does not work, please upload the following file: 'TRANSIENT.lst' from your Modflow output folder.

Good luck

Dor

**Reply by weixiaoxun** [reply]
Dear Professor @dof1985 ,

Thank you very much for your previous response. I have added use_complex_solver_for_modflow=True to the ini file, and the simulation period is set from 1981 to 2004. I thought this would resolve the issue, but unfortunately, I encountered the same error in 2002.

I have attached the TRANSIENT.lst （Since the file type is not supported, I have converted it to a .txt file） to this email for your review. Could you kindly help me understand the cause of the error?

Thank you once again for your guidance and support.
 
Best regards,
Wei xiaoxun
[TRANSIENT_lst_into_txt.txt](https://github.com/user-attachments/files/17095144/TRANSIENT_lst_into_txt.txt)

**Comment by dof1985** [comment]
Hi @weixiaoxun 

The .lst file does not show anything strange.
The error you experience implies that MODFLOW's caluculation of the Richardson equations could not converge. From my experience it may sometime be the result of some random instabilities; in other cases it may be associated with 'problematic' inputs, or states of some of the modeled stocks.

I would recommend to run it again. Also save the following daily maps: head, groundwater_storage_available, sumed_sum_gwRecharge, modfPumpingM; if it crashes again we can inspect these layers

Good luck

be we can find something in different maps.

**Reply by weixiaoxun** [reply]
Dear @dof1985 ,

I hope this email finds you well.

I would like to sincerely thank you for your valuable advice. With your guidance, I was able to successfully run CWatM-ModFlow for the first time. However, after generating the GWSA output, the graph I obtained turned out to be a smooth curve, which I believe should actually be more fluctuating.

I have attached two images for your reference: the first one shows my ini file, and the second is the GWSA output graph I generated. I would greatly appreciate it if you could kindly review them and provide me with further guidance.

Thank you again for your time and support!

Best regards,
Wei xiaoxun
[settings_CWatM_template_30min_chinanorth.txt](https://github.com/user-attachments/files/17106470/settings_CWatM_template_30min_chinanorth.txt)
![微信截图_20240924090748](https://github.com/user-attachments/assets/8554de47-63ca-4f03-8c75-68d7fde5a784)

**Comment by dof1985** [comment]
Dear @weixiaoxun 

Overall, this is a new topic and I would propose to start a new discussion (I will close this discussion now). Anyway, here are some few tips. 
Your results seem to make sense for North China.
1/ I do see some fluctuations, but if you are looking for seasonal pattern, this may be hidden due to the scale. You can either zoom in to observe only one year, or detrend the data to observe these patterns.
2/ You can also use this tutorial to compare the decrease in groundwater levels with GRACE data: https://www.youtube.com/watch?v=aWQa0ZdYaWY

Good luck
Dor
